finagle-netty4/src/test/scala/com/twitter/finagle/netty4/Netty4SslTest.scala

```diff
@@ -0,0 +1,127 @@
+      Netty4Transporter.raw[String, String](pipeline => {
+        pipeline.addLast(new DelimiterBasedFrameDecoder(8192, Delimiters.lineDelimiter(): _*))
+        pipeline.addLast(new StringDecoder())
+        pipeline.addLast(new StringEncoder())
+      }, addr, params)
+
+    override protected def newDispatcher(transport: Transport[String, String]): Service[String, String] =
+      new SerialClientDispatcher(transport)
+
+    override protected def copy1(stack: Stack[ServiceFactory[String, String]], params: Params) =
+      copy(clientCert, serverCert, params, stack)
+  }
+}
+
+class Netty4SslTest extends FunSuite with Eventually with IntegrationPatience {
+
+  class Ctx {
+    val serverCert = new SelfSignedCertificate("example.server.com")
+    val clientCert = new SelfSignedCertificate("example.client.com")
+    val allocator = io.netty.buffer.UnpooledByteBufAllocator.DEFAULT
+
+    private object StringServerInit extends (ChannelPipeline => Unit) {
+      def apply(pipeline: ChannelPipeline): Unit = {
+        pipeline.addLast("line", new DelimiterBasedFrameDecoder(100, Delimiters.lineDelimiter(): _*))
+        pipeline.addLast("stringDecoder", new StringDecoder(UTF_8))
+        pipeline.addLast("stringEncoder", new StringEncoder(UTF_8))
+      }
+    }
+
+    val server = {
+      val service =
+        new Service[String, String] {
+          override def apply(request: String): Future[String] = {
+            Future.value(
+              Transport.peerCertificate match {
+                case Some(_) => "OK\n"
+                case None => "ERROR\n"
+              }
+            )
+          }
+        }
+
+      val p = Params.empty +
+        ServerSsl(Some(SslServerConfiguration(clientAuth = ClientAuth.Needed))) +
+        SslServerEngineFactory.Param(new SslServerEngineFactory {
+          override def apply(config: SslServerConfiguration): Engine = {
+            val context = SslContextBuilder.forServer(serverCert.key(), serverCert.cert()).trustManager(clientCert.cert()).build()
+            val engine = context.newEngine(allocator)
+            engine.setNeedClientAuth(true)
+            new Engine(engine)
+          }
+        }) +
+        Label("test")
+      val listener = Netty4Listener[String, String](StringServerInit, p)
+      val serveTransport = (t: Transport[String, String]) => {
+        if (t.peerCertificate.isEmpty) throw new IllegalStateException("No peer certificate in transport")
+        new SerialServerDispatcher(t, service)
+      }
+      listener.listen(new InetSocketAddress(InetAddress.getLoopbackAddress, 0))(serveTransport(_))
+    }
+
+    val client = {
+      val addr = server.boundAddress.asInstanceOf[InetSocketAddress]
+      new finagle.netty4.Netty4SslTest.Client(clientCert, serverCert)
+        .withTransport.tls(SslClientConfiguration(), new SslClientEngineFactory {
+        override def apply(address: Address, config: SslClientConfiguration): Engine = {
```

Comment:
You should be able to just use the Client Configuration here. Something like ...

.withTransport.tls(SslClientConfiguration(
  keyCredentials = KeyCredentials.CertAndKey(new File(clientCert.cert()), new File(clientCert.key())),
  trustCredentials = TrustCredentials.CertCollection(new File(serverCert.cert())))

and you could optionally include the Netty4SslClientEngineFactory as the second parameter.